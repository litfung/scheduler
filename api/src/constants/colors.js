module.exports = {
  group_color_list: [
    '#f03434', // Redish
    '#4286f4', // Blueish
    '#f7ca18', // Yellowish
    '#ff9b17', // Orangish
    '#00bc60' // Greenish
  ],
};