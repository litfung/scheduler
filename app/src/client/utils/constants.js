export const month_name_dictionary = {
  0: 'January',
  1: 'February',
  2: 'March',
  3: 'April',
  4: 'May',
  5: 'June',
  6: 'July',
  7: 'August',
  8: 'September',
  9: 'October',
  10: 'November',
  11: 'December',
};

export const day_name_dictionary = [
  'SUN', 'MON', 'TUE', 'WED', 'THU', 'FRI', 'SAT'
];

export const builtin_color_list = ['#f03434', '#59abe3', '#8c14fc', '#2ecc71', '#f7ca18'];