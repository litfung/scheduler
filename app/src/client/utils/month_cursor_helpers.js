
import moment from 'moment';

export const funcHandleYear = (prevYear, prevMonth, bool) => {
  if (prevMonth !== 1 && prevMonth !== 12) return prevYear;
  else if (prevMonth === 12) return bool ? prevYear + 1 : prevYear;
  else if (prevMonth === 1) return bool ? prevYear : prevYear - 1
}

export const funcHandleMonth = (prevMonth, bool) => {
  if (prevMonth === 1) return bool ? prevMonth + 1 : 12;
  else if (prevMonth === 12) return bool ? 1 : prevMonth - 1;
  else return bool ? prevMonth + 1 : prevMonth - 1;
}

export const generateMomentMonth = (year, month) => {
  let temp_text;
  if (month.toString().length === 1) temp_text = `${year}-0${month}`;
  else temp_text = `${year}-${month}`;
  return moment(temp_text);
}

export const formatISOStringForMoment = (year, month, date) => {
  const myMonth = (month.toString().length === 1) ? `0${month}` : `${month}`;
  const myDate = (date.toString().length === 1) ? `0${date}` : `${date}`;
  return `${year}-${myMonth}-${myDate}`;
};