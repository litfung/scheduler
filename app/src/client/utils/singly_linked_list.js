// Data Structures - Singly Linked List

class Node {
  constructor(value) {
    this.value = value;
    this.next = null;
  }
}

class SinglyLinkedList {
  constructor() {
    this.head = null;
    this.tail = null;
    this.length = 0;
  }

  push(value) {
    const newNode = new Node(value);
    if (!this.head) {
      this.head = newNode;
      this.tail = this.head;
    } else {
      this.tail.next = newNode;
      this.tail = newNode;
    }
    this.length++;
    return this;
  }

  pop() {
    if (!this.head) return undefined;

    let current = this.head;
    let prev = current;
    while(current.next) {
      prev = current;
      current = current.next;
    }
    this.tail = new Node(prev.value);
    this.length--;
    if (this.length === 0) {
      this.head = null;
      this.tail = null;
    }
    return current;
  }

  shift() {
    if (!this.head) return undefined;

    const temp = this.head;
    this.head = temp.next;
    
    if (this.length <= 1) this.tail = null;
    if (this.length > 0) this.length--;
    
    return temp;
  }

  unshift(value) {
    if (!value && value !== 0) return undefined;

    if (this.head) {
      const temp = this.head;
      this.head = new Node(value);
      this.head.next = temp;
    } else {
      this.head = new Node(value);
      this.tail = this.head;
    }
    
    this.length++;
    return this;
  }

  get(index) {
    let current = this.head;
    let counter = 0;

    while (current) {
      if (index === counter) return current;
      current = current.next;
      counter++;
    }

    return undefined;
  }

  set(index, value) {
    const prevNode = this.get(index);
    if (prevNode) {
      prevNode.value = value;
      return true;
    }
    return false;
  }

  insert(index, value) {
    if (index < 0 || index >= this.length || !value) return undefined;
    if (this.length === 0) return this.push(value);

    const newNode = new Node(value);

    if (index === 0) {
      newNode.next = this.head;
      this.head = newNode;
      this.length++;
      return this;
    }

    const currentNode = this.get(index);
    const prevNode = this.get(index - 1);
    newNode.next = currentNode;
    prevNode.next = newNode;
    this.length++;
    
    return this; 
  }

  remove(index) {
    if (index < 0 || index >= this.length) return undefined;

    if (index === (this.length - 1)) return !!this.pop();
    if (index === 0) return !!this.shift();

    const currentNode = this.get(index);
    const prevNode = this.get(index - 1);
    prevNode.next = currentNode.next;
    this.length--;
    
    return this;

  }

  reverse() {
    let node = this.head;
    this.head = this.tail;
    this.tail = node;

    let next;
    let prev = null;

    for (var i = 0; i < this.length; i++) {
      next = node.next;
      node.next = prev;
      prev = node;
      node = next;
    }

    return this;
  }
}
