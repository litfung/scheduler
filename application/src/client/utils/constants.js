export const month_name_dictionary = {
  0: 'January',
  1: 'February',
  2: 'March',
  3: 'April',
  4: 'May',
  5: 'June',
  6: 'July',
  7: 'August',
  8: 'September',
  9: 'October',
  10: 'November',
  11: 'December',
};

export const day_name_dictionary = [
  'SUN', 'MON', 'TUE', 'WED', 'THU', 'FRI', 'SAT'
];

export const builtin_color_list = [
  '#f03434', // Redish
  '#088cfb', // Blueish
  '#f7ca18', // Yellowish
  '#ff9b17', // Orangish
  '#00bc60' // Greenish
];

// export const intro_paragraph = 'It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout. The point of using Lorem Ipsum is that it has a more-or-less normal distribution of letters, as opposed to using \'Content here, content here\', making it look like readable English.';
export const intro_paragraph = `Schedule your work in style. From work to groceries a task scheduler is the best way to get things done. I'm not a very creative person and I can't think of more lines to add, so here's some gibberish -  Desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.`;