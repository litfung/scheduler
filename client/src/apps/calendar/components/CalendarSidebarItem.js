import React, { useState, useEffect } from 'react';
import { FaEllipsisV } from 'react-icons/fa';

import EventDoneIndicator from '../../_common/components/EventDoneIndicator';
import OptionsGroup from '../../_common/components/OptionsGroup';

const CalendarSidebarItem = (props) => {
	const { group, handleGroupRename, handleGroupDelete, handleGroupVisiblity } = props;

	const [ title, setTitle ] = useState(group.title);
	const [ isOnCalendar, setIsOnCalendar ] = useState(group._isOnCalendar);
	const [ inputDis, setInputDis ] = useState(true);
	const [ dropdown, setDropdown ] = useState(false);
	const [ screenX, setScreenX ] = useState(null);
	const [ screenY, setScreenY ] = useState(null);
	const [ winHeightDiffer, setWinHeightDiffer ] = useState(0);

	useEffect(
		() => {
			setIsOnCalendar(group._isOnCalendar);
		},
		[ group._isOnCalendar ]
	);

	const showDropdown = (x, y, whd) => {
		setWinHeightDiffer(whd);
		setScreenX(x);
		setScreenY(y);
		setDropdown(true);
	};

	const hideDropdown = () => {
		setDropdown(false);
		setScreenX(null);
		setScreenY(null);
	};

	return (
		<div className="CalendarSidebarItem-c-00">
			<div style={{ display: 'flex', flexDirection: 'row', alignItems: 'center' }}>
				<EventDoneIndicator
					_isDone={isOnCalendar}
					hex_color={group.hex_color}
					patchFunction={() => {
						handleGroupVisiblity(group._id, !isOnCalendar);
						setIsOnCalendar(!isOnCalendar);
					}}
				/>
				<form
					onSubmit={(e) => {
						e.preventDefault();
						handleGroupRename({ groupId: group._id, title, hex_color: group.hex_color });
					}}
				>
					<input
						id={`CalendarSidebarItem-Input-xx-${group._id}`}
						className="Theme-Input-Underline-OnFoucs-99"
						disabled={inputDis}
						name="title"
						autoComplete="off"
						value={title}
						onChange={(e) => setTitle(e.target.value)}
						onBlur={() => {
							setTitle(group.title);
							setInputDis(true);
						}}
					/>
				</form>
			</div>

			<OptionsGroup
				pathName="calendar"
				visible={dropdown}
				group={group}
				winHeightDiffer={winHeightDiffer}
				positionObj={{
					left: `calc(${screenX}px - 8px)`,
					bottom: `calc(100vh - ${screenY - winHeightDiffer + 8}px)`
				}}
				handleGroupDelete={() => handleGroupDelete(group._id)}
				showDropdown={showDropdown}
				hideDropdown={hideDropdown}
				onRenameClick={async () => {
					await setInputDis(false);
					document.querySelector(`#CalendarSidebarItem-Input-xx-${group._id}`).focus();
				}}
				// hex_color={group.hex_color}
			>
				<button
					className="CalendarSidebarItem-Options-Btn-02 Flex-Class-Row-Center"
					onClick={() => setDropdown(true)}
				>
					<FaEllipsisV />
				</button>
			</OptionsGroup>
		</div>
	);
};

export default CalendarSidebarItem;
