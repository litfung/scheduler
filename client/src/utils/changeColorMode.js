import { setCookie } from './cookie_funcs';

const myAppThemes = {
	lightOnly: {
		'--danger-red-color': '#f03434',
		'--safe-green-color': '#019031',
		'--background-color': 'white',
		'--border-color': '#d8d8d8',
		'--hover-mint-light': '#f3f3f3',
		'--hover-mint-medium': '#f3f3f3',
		'--hover-mint-deep': '#dddddd',
		'--text-color': '#424242',
		'--text-color-light': '#707070',
		'--text-color-light-2': '#b8b8b8',
		'--text-color-light-3': '#707070',
		'--modal-mask-color': 'rgba(0, 0, 0, 0.3)'
	},
	darkOnly: {
		'--danger-red-color': '#f03434',
		'--safe-green-color': '#019031',
		'--background-color': '#151a1d',
		'--border-color': '#292f33',
		'--hover-mint-light': '#1c2225',
		'--hover-mint-medium': '#1c2225',
		'--hover-mint-deep': '#292f33',
		'--theme-color-deep': '#0071e2',
		'--text-color': 'white',
		'--text-color-light': 'white',
		'--text-color-light-2': '#999999',
		'--text-color-light-3': '#707070',
		'--modal-mask-color': 'rgba(255, 255, 255, 0.1)'
	}
};

export default (name) => {
	if (!name) name = 'lightOnly';

	if (window) {
		const html = document.getElementsByTagName('html')[0];
		const obj = myAppThemes[name];

		Object.keys(obj).map((key) => {
			html.style.setProperty(key, obj[key]);
		});

		setCookie('myAppColorMode', name, 365); // 'name' is value here
	}
};
