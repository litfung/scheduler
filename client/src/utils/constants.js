export const month_name_dictionary = {
	0: 'January',
	1: 'February',
	2: 'March',
	3: 'April',
	4: 'May',
	5: 'June',
	6: 'July',
	7: 'August',
	8: 'September',
	9: 'October',
	10: 'November',
	11: 'December'
};

export const day_name_dictionary = [ 'SUN', 'MON', 'TUE', 'WED', 'THU', 'FRI', 'SAT' ];

export const builtin_color_list = [
	'#f03434', // Redish
	'#088cfb', // Blueish
	'#f7ca18', // Yellowish
	'#ff9b17', // Orangish
	'#00bc60' // Greenish
];

export const app_theme_options = [
	{ title: 'Red', hex_color: '#f03434' },
	{ title: 'Blue', hex_color: '#088cfb' },
	{ title: 'Yellow', hex_color: '#f7ca18' },
	{ title: 'Orange', hex_color: '#ff9b17' },
	{ title: 'Green', hex_color: '#00bc60' }
];
