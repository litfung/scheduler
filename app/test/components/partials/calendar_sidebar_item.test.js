import React from 'react';
import { shallow } from 'enzyme';

import { builtin_color_list } from '../../../src/client/utils/constants'
import groups from '../../__data__/groups';

import CalendarSidebarItem from '../../../src/client/components/partials/calendar/calendar_sidebar_item';

let wrapper, changeColorFunc, asyncEditGroup, asyncDeleteGroup;

beforeEach(() => {
  changeColorFunc = jest.fn();
  asyncEditGroup = jest.fn();
  asyncDeleteGroup = jest.fn();

  wrapper = shallow(
    <CalendarSidebarItem
      index={0}
      group={groups[0]}
      asyncEditGroup={asyncEditGroup}
      asyncDeleteGroup={asyncDeleteGroup}
      color_options={[ ...builtin_color_list ]}
      changeColorFunc={changeColorFunc}
    />
  );
});

test('Snapshot for CalendarSidebarItem Component', () => {
  expect(wrapper).toMatchSnapshot();
});
