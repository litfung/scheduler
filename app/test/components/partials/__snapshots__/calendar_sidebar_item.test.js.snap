// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot for CalendarSidebarItem Component 1`] = `
<div
  className="calendar-sidebar-item-000"
>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <EventDoneIndicator
      _isDone={true}
      hex_color="#4286f4"
      hide_tick_on_false={true}
      patchFunction={[Function]}
      squareShaped={true}
    />
    <form
      className="calendar-sidebar-item-title-form-002"
      onSubmit={[Function]}
    >
      <input
        autoComplete="off"
        disabled={true}
        id="calendar-sidebar-item-input-inside-form-5c373f59d64892352c703256"
        name="title"
        onBlur={[Function]}
        onChange={[Function]}
        value=". . ."
      />
    </form>
  </div>
  <GroupOptDropdownComp
    asyncDeleteGroup={[MockFunction]}
    asyncEditGroup={[MockFunction]}
    changeColorFunc={[MockFunction]}
    color_options={
      Array [
        "#f03434",
        "#4286f4",
        "#f7ca18",
        "#ff9b17",
        "#00b16a",
      ]
    }
    color_panel={
      Object {
        "screenX": null,
        "screenY": null,
        "visible": false,
      }
    }
    dropdown_visible={false}
    group={
      Object {
        "__v": 0,
        "_creator": "5c373f59d64892352c703255",
        "_id": "5c373f59d64892352c703256",
        "_isDeleted": false,
        "_isOnCalendar": true,
        "_isPermanent": true,
        "_rank": 0,
        "hex_color": "#4286f4",
        "title": "Tasks",
      }
    }
    index={0}
    input_disable={true}
    onRenameClick={[Function]}
    positionObj={
      Object {
        "bottom": "calc(100vh - 8px)",
        "left": "calc(nullpx - 8px)",
      }
    }
    screenX={null}
    screenY={null}
    setParentState={[Function]}
    subPositionObj={
      Object {
        "bottom": "calc(100vh - 8px)",
        "left": "calc(nullpx - 8px)",
      }
    }
    title=". . ."
    windowHeightDiff={0}
  >
    <button
      className="calendar-sidebar-item-options-btn"
      name="More Options"
      onClick={[Function]}
    >
      <FaEllipsisV />
    </button>
  </GroupOptDropdownComp>
</div>
`;
